operator/v1/types_ingress.go

```diff
@@ -233,8 +233,36 @@ type LoadBalancerStrategy struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Scope LoadBalancerScope `json:"scope"`
+	// aws provides configuration settings that are specific to AWS
+	// load balancers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	AWS *AWSLoadBalancerParameters `json:"aws,omitempty"`
 }
 
+// AWSLoadBalancerParameters provides configuration settings that are
+// specific to AWS load balancers.
+type AWSLoadBalancerParameters struct {
+	// type is the type of AWS load balancer to instantiate for an ingress
+	// controller.
+	//
+	// If unspecified, "Classic" is used.
+	//
+	// +optional
+	Type AWSLoadBalancerType `json:"type,omitempty"`
```

Comment:
Because this field is optional and named "type", does that leave open the option of changing it to a union discriminator in the future?  I'm not sure whether or not that is something we are likely to need, but I'd like to know whether it leaves us the option.

Reply:
Commit `61d6d39` makes `AWSLoadBalancerParameters` a union type with the `Type` field as the union discriminator.

Reply:
All the examples I found when searching for "idiomatic golang enum declaration" use `const` and not `var`.